addons/project/models/project.py

```diff
@@ -456,6 +457,13 @@ def map_tasks(self, new_project_id):
             new_child_ids = [old_to_new_tasks[child.id] for child in task.child_ids if child.id in old_to_new_tasks]
             tasks.browse(new_child_ids).write({'parent_id': new_task.id})
             old_to_new_tasks[task.id] = new_task.id
+            if task.allow_task_dependencies:
+                depend_on_ids = [t.id if t not in all_tasks else old_to_new_tasks.get(t.id, None) for t in
+                                 task.depend_on_ids]
```

Comment:
I quite don't understand why you are using `old_to_new_tasks.get(t.id, None)` as this would only happen if the record you are trying to get the copy from will be handled later in the `for task in all_tasks:`. But in such a case we should wait for the record to get copied and then add it to the dependencies of the task.

So we should either make a new loop after `for task in all_tasks:` in order to treat the `task.allow_task_dependencies` case, or find a way to add records in old_to_new_tasks from within the `if task.allow_task_dependencies`and avoid treating them after in the `for task in all_tasks:` when they will appear.

I hope you get the issue, this is maybe not so clear and easy to explain. Do no hesitate to contact me would you have any question.

Thanks!